map/src/map/components/layers/PlanRouteLayer.js

```diff
@@ -0,0 +1,153 @@
+import React, {useContext, useEffect, useState} from "react";
+import AppContext from "../../../context/AppContext";
+import {useMap} from "react-leaflet";
+import PanelButtons from "../PanelButtons";
+import Utils from "../../../util/Utils";
+import SaveRouteDialog from "../SaveRouteDialog";
+
+
+export default function PlanRouteLayer() {
+    const ctx = useContext(AppContext);
+    const map = useMap();
+
+    const [openSaveDialog, setOpenSaveDialog] = useState(false);
+    const [openPanelButtons, setOpenPanelButtons] = useState(false);
+    const [prevCreatedTracks, setPrevCreatedTracks] = useState(null);
+
+    if (ctx.currentlyEditTrack) {
+        map.on("editable:drawing:clicked", e => {
+            ctx.currentlyEditTrackDispatch({
+                type: 'click',
+                e: e
+            })
+        });
+
+        map.on("editable:vertex:deleted", e => {
+            ctx.currentlyEditTrackDispatch({
+                type: 'deletedClick',
+                e: e
+            })
+        });
+
+        map.on("editable:vertex:dragend", e => {
+            ctx.currentlyEditTrackDispatch({
+                type: 'dragendClick',
+            })
+        });
+    }
+
+    useEffect(() => {
+        let selectedTrack = ctx.createdTracks.find(t => t.selected === true);
+        if (selectedTrack) {
+            setPrevCreatedTracks(ctx.createdTracks);
+            setOpenPanelButtons(true);
+            let newRouteLayer;
+            let newPoints = [];
+            if (selectedTrack.points && selectedTrack.points.length > 0) {
+                deleteOldRoute(map);
+
+                if (ctx.currentlyEditTrack === null) {
+                    ctx.currentlyEditTrackDispatch({
+                        type: 'createTrack',
+                    })
+                }
+
+                selectedTrack.points.forEach(function (item) {
+                    newPoints.push({lat: item.lat, lng: item.lng});
+                });
+                newRouteLayer = map.editTools.addPolylineByPoints(newPoints);
+                map.fitBounds(newRouteLayer && newRouteLayer._bounds);
+
+                ctx.currentlyEditTrackDispatch({
+                    type: 'showTrack',
+                    track: selectedTrack,
+                    pointsList: structuredClone(Utils.getPointsDist(selectedTrack.points)),
+                    routeLayer: newRouteLayer
+                })
+            }
+        } else if (prevCreatedTracks && prevCreatedTracks.length >= ctx.createdTracks.length) {
+            deleteOldRoute(map);
+        }
+    }, [ctx.createdTracks, ctx.setCreatedTracks]);
+
+
+    useEffect(() => {
+        if (ctx.createNewTrack) {
+            if (ctx.currentlyEditTrack) {
+                deleteOldRoute(map);
```

Comment:
this is the only method with map and could be handled with useEffect(...state.clean ?)